src/plugins/rv-packages/otio_reader/clipHook.py

```diff
@@ -8,4 +8,25 @@
 
 # this is an example for clip that can be removed or augmented as required
 def hook_function(in_timeline, argument_map=None):
+    if isinstance(in_timeline.media_reference, otio.schema.ExternalReference):
```

Comment:
Thinking about how to implement batching... perhaps the OpenAssetIO package could include its own OTIO manifest.json in OTIO_PLUGIN_MANIFEST_PATH and insert its own hooks? To batch up requests I'd probably need to register pre_hook_Timeline and do a separate walk of the clips?

Reply:
I think that would work yes. If you look in the method `_get_global_transform` in the otio_reader it is doing something similar.  It's looping through all clips at the timeline level to get their spatial bounds before importing them.

You could loop through the same in a pre-Timeline hook to get all the references and then resolve them in a single call before the clips are actually imported.  There's a context object that gets passed along to each hook for its lifetime, so you could add the results to the context object for the Timeline, then in each clip you simply do a lookup in the context object.

Reply:
Since there's some demo specific stuff in there, I think it makes sense to work off a branch.  Then when we're done we can do a PR to the main repo with any parts we might want to make officially part of OpenRV.  I'll start one tomorrow.

Reply:
Cool! That gives us a couple kicks at it before we decide on the best separation of concerns

Reply:
It wouldn't let me edit the PR to change repos (only branches on the same repo), so I opened a new PR to merge into a new branch on OpenRV:
https://github.com/AcademySoftwareFoundation/OpenRV/pull/114

Reply:
Great! MovieLabs has signed the company-level CLA now so we should be good to go.